node/sync/src/block_sync.rs

```diff
@@ -1518,7 +1518,9 @@ mod tests {
         }
 
         // Duplicate a new sync module with a different height to simulate block advancement.
-        let ledger_height = rng.gen_range(0..locator_height);
+        // This range needs to be inclusive, so that the range is never empty,
+        // even with a locator height of 0.
+        let ledger_height = rng.gen_range(0..=locator_height);
```

Comment:
@raychu86 The previous CI run for my branch failed this test, and could reproduce it failing every 100 times or so on my local machine. It is unrelated to the PR, but I think we should just fix it now.

Could you check if the fix is correct? It looks like you originally wrote the test.

Reply:
This is correct, thanks for finding this issue.

I assume this would create an empty range and fail whenever `let locator_height = rng.gen_range(0..50);` resolved to 0.